services/app-api/resolvers/fetchSubmission2.test.ts

```diff
@@ -0,0 +1,281 @@
+            query: FETCH_SUBMISSION_2,
+            variables: { input: fetchInput },
+        })
+
+        expect(resubmitResult.errors).toBeUndefined()
+
+        expect(resubmitResult.data?.fetchSubmission2.submission.status).toEqual('RESUBMITTED')
+        expect(resubmitResult.data?.fetchSubmission2.submission.submittedAt).toEqual(today)
+
+
+    })
+
+    it('a different user from the same state can fetch the draft', async () => {
+        const server = await constructTestPostgresServer()
+
+        // First, create a new submission
+        const stateSubmission = await createTestStateSubmission(server)
+
+        const createdID = stateSubmission.id
+
+        // then see if we can fetch that same submission
+        const input = {
+            submissionID: createdID,
+        }
+
+        // setup a server with a different user
+        const otherUserServer = await constructTestPostgresServer({
+            context: {
+                user: {
+                    name: 'Aang',
+                    state_code: 'FL',
+                    role: 'STATE_USER',
+                    email: 'aang@mn.gov',
+                },
+            },
+        })
+
+        const result = await otherUserServer.executeOperation({
+            query: FETCH_SUBMISSION_2,
+            variables: { input },
+        })
+
+        expect(result.errors).toBeUndefined()
+
+        expect(result.data?.fetchSubmission2.submission).toBeDefined()
+        expect(result.data?.fetchSubmission2.submission).not.toBeNull()
+    })
+
+    it('returns an error if you are requesting for a different state (403)', async () => {
+        const server = await constructTestPostgresServer()
+
+        // First, create a new submission
+        const stateSubmission = await createTestStateSubmission(server)
+
+        const createdID = stateSubmission.id
+
+        // then see if we can fetch that same submission
+        const input = {
+            submissionID: createdID,
+        }
+
+        // setup a server with a different user
+        const otherUserServer = await constructTestPostgresServer({
+            context: {
+                user: {
+                    name: 'Aang',
+                    state_code: 'VA',
+                    role: 'STATE_USER',
+                    email: 'aang@va.gov',
+                },
+            },
+        })
+
+        const result = await otherUserServer.executeOperation({
+            query: FETCH_SUBMISSION_2,
+            variables: { input },
+        })
+
+        expect(result.errors).toBeDefined()
+        if (result.errors === undefined) {
+            throw new Error('annoying jest typing behavior')
+        }
+        expect(result.errors?.length).toEqual(1)
+        const resultErr = result.errors[0]
+
+        expect(resultErr?.message).toEqual(
+            'user not authorized to fetch data from a different state'
+        )
+        expect(resultErr?.extensions?.code).toEqual('FORBIDDEN')
+    })
+
```

Comment:
how about a test for what should happen if a submission at that ID does not exist? Or is that handled somewhere else.

Reply:
This test caught a bug!